### PR TITLE
Python Runner support + pytest and unittest examples

### PR DESCRIPTION
**Pull request by spider-sauce**
## Notes:
* Added `.circleci` directory for [runner support](https://circleci.com/gh/saucelabs-training/demo-python/34)
* CircleCI requires Sauce Connect Proxy, so on-boarding UI will need to explicitly state for users NOT testing behind private/corporate network to comment out the following line in each applicable test:
```
'tunnelIdentifier': 'demo-python-tunnel'
```
* Added both `pytest` and `unittest` examples so that when/if we decide to implement framework dropdown options the code is ready to go
* Module1 and Module2 code still exist with negligible changes including `tunnelIdentifier`, some updated comments, and their new location: `~/on-boarding-modules/python-examples`
* Module3 and Module4 require frameworks so the options are in the following directories:
   * `~/on-boarding-modules/pytest-examples`
   * `~/on-boarding-modules/unittest-examples`

### CODE REVIEW

File: on-boarding-modules/pytest-examples/conftest.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 import os
 from selenium import webdriver
 
+# skipping the first file
+collect_ignore = ["test_module1_pytest.py"]
```
Comment (richard-sl) [inline on-boarding-modules/pytest-examples/conftest.py]: Why ignore module1 test?

File: on-boarding-modules/pytest-examples/test_module1_pytest.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Selenium 3.14+ doesn't enable certificate checking
+import pytest
+from selenium import webdriver
+from _pytest.runner import runtestprotocol
+import urllib3
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+# @pytest-examples.fixture
+@pytest.mark.skip(reason="Update Sauce Creds")
```
Comment (richard-sl) [inline on-boarding-modules/pytest-examples/test_module1_pytest.py]: Is there some coverage of demonstrating pytest that this is here? I am not sure about conflating pytest demonstration within module1.

**Comment by richard-sl** [comment]
>  urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning) 

That does not seem like something we want to be coaching users on. Could we do better to at least only trust in regards to Sauce certificates? Globally patching urllib opens up more than needed. 

> CircleCI requires Sauce Connect Proxy, so on-boarding UI will need to explicitly state for users NOT testing behind private/corporate network to comment out the following line in each applicable test:

Why does CircleCI require Sauce Connect? And it needs 3 tunnels? The demo app should be publicly available on www.saucedemo.com :(

Why default configuration? I would suggest something roughly like `desired_capabilities['tunnelIdentifier'] = os.environ['CI_TUNNEL_ID'] || None` so that we don't have to tell users to manually remove something. Then for circle CI could inject the tunnel name through envvar as needed. 

> driver.get("http://www.saucedemo.com")

Why not `driver.get("https://...")`?

File: on-boarding-modules/pytest-examples/conftest.py (hunk 1 above)
Comment (spider-sauce) [reply on-boarding-modules/pytest-examples/conftest.py]: Because that's the module that requires users to hard code `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY`. It's basically the same test as module2

**Comment by spider-sauce** [comment]
`urllib3` is an ongoing issue with a lot of our python examples. Working with Josh to resolve that, for now that's why we used http in the SUT URL

In reference to `desired_capabilities['tunnelIdentifier'] = os.environ['CI_TUNNEL_ID'] || None`, we would still need to tell users to create a local envar called `CI_TUNNEL_ID` and have them set it to null. Otherwise when they test locally they'll get this error:
```
os.environ['CI_TUNNEL_ID']
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/os.py", line 678, in __getitem__
    raise KeyError(key) from None
KeyError: 'CI_TUNNEL_ID'
```
So basically it's either they remove something or create an empty variable. The empty variable approach might be good for later modules when they eventually setup Sauce Connect

Also working with Josh to see if there's a better way that's non invasive to users' learning.

File: on-boarding-modules/pytest-examples/test_module1_pytest.py (hunk 2 above)
Comment (spider-sauce) [reply on-boarding-modules/pytest-examples/test_module1_pytest.py]: That flag was for skipping the test for CI, I could just update the `config.yml` to explicitly leave out the test during the build step and then remove the pytest flag here
